From 2055a46b396e272616c0b2273903e02c3b49a2ff Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Tue, 10 Nov 2020 16:33:27 +0000
Subject: [PATCH] src/cmd/dist/buildgo.go: do not hardcode host compilers into
 target binaries

These come from $CC/$CXX on the build host and are not useful on targets;
additionally as they contain host specific paths, this helps reproducibility.

Upstream-Status: Inappropriate [needs upstream discussion]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 src/cmd/dist/buildgo.go | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/cmd/dist/buildgo.go b/src/cmd/dist/buildgo.go
index caafc13..4eb1c96 100644
--- a/src/cmd/dist/buildgo.go
+++ b/src/cmd/dist/buildgo.go
@@ -34,8 +34,8 @@ func mkzdefaultcc(dir, file string) {
 		fmt.Fprintf(&buf, "package cfg\n")
 		fmt.Fprintln(&buf)
 		fmt.Fprintf(&buf, "const DefaultPkgConfig = `%s`\n", defaultpkgconfig)
-		buf.WriteString(defaultCCFunc("DefaultCC", defaultcc))
-		buf.WriteString(defaultCCFunc("DefaultCXX", defaultcxx))
+		buf.WriteString(defaultCCFunc("DefaultCC", map[string]string{"":"gcc"}))
+		buf.WriteString(defaultCCFunc("DefaultCXX", map[string]string{"":"g++"}))
 		writefile(buf.String(), file, writeSkipSame)
 		return
 	}
@@ -46,8 +46,8 @@ func mkzdefaultcc(dir, file string) {
 	fmt.Fprintf(&buf, "package main\n")
 	fmt.Fprintln(&buf)
 	fmt.Fprintf(&buf, "const defaultPkgConfig = `%s`\n", defaultpkgconfig)
-	buf.WriteString(defaultCCFunc("defaultCC", defaultcc))
-	buf.WriteString(defaultCCFunc("defaultCXX", defaultcxx))
+	buf.WriteString(defaultCCFunc("defaultCC", map[string]string{"":"gcc"}))
+	buf.WriteString(defaultCCFunc("defaultCXX", map[string]string{"":"g++"}))
 	writefile(buf.String(), file, writeSkipSame)
 }
 
